.travis.yml

```diff
@@ -3,7 +3,7 @@ go:
   - 1.9
 dist: trusty
 install:
-  - sudo apt-get install -y build-essential swig oracle-java8-installer g++ python3-dev
+  - sudo apt-get install -y build-essential swig oracle-java8-installer g++ python3-dev mono-mcs
```

Comment:
Just for my understanding : the dependency on mono here is just required by swig, right?

Can you confirm the generated cs code can be consumed by any CLR implementation (_regular_.net, .net Core or mono)?

Reply:
No it cannot, the final step of the making of the dll is made by mono (mcs). If we want to to do this with regular .net or something else we have to change this.
I am not able to do this as I have absolutely no knowledge about .net and windows in general so I think we should do an issue if we want to include regular .net compiler and how to do it etc.
The goal of this PR is to get a draft of the C# SDK and then get feedbacks from people who knows it to then get a stable C# SDK

Reply:
ok, fine by me. I'll try to have a look at if the dll can be consumed by official .Net CLR (I doubt .Net core will) if I find time.